src/main/java/vmware/samples/vcenter/vm/power/PowerLifeCycle.java

```diff
@@ -0,0 +1,119 @@
+package vmware.samples.vcenter.vm.power;
+
+import java.util.Collections;
+import java.util.List;
+
+import org.apache.commons.cli.Option;
+
+import com.vmware.vcenter.vm.Power;
+import com.vmware.vcenter.vm.PowerTypes;
+
+import vmware.samples.common.SamplesAbstractBase;
+import vmware.samples.vcenter.helpers.VmHelper;
+
+/**
+ * Description: Demonstrates the Power Life Cycle of a VM
+ *
+ * Author: VMware, Inc.
+ * Sample Prerequisites: The sample needs an existing VM
+ */
+public class PowerLifeCycle extends SamplesAbstractBase{
+
+    private String vmName;
+    private Power vmPowerService;
+    private String vmId;
+
+    @Override
+    protected void parseArgs(String[] args) {
+        // Parse the command line options or use config file
+        Option vmNameOption = Option.builder()
+            .longOpt("vmname")
+            .desc("Name of the VM on which the power operations"
+            		+ " would be performed")
+            .required(true)
+            .hasArg()
+            .argName("VM NAME")
+            .build();
+        List<Option> optionList = Collections.singletonList(vmNameOption);
+        super.parseArgs(optionList, args);
+        this.vmName =  (String) parsedOptions.get("vmname");        
+    }
+
+    @Override
+    protected void setup() throws Exception {
+        vmId = VmHelper.getVM(vapiAuthHelper.getStubFactory(),
```

Comment:
Please refer all instance variables with "this." for consistency with other samples.

Reply:
Done!!